viper/parser/expr.py

```diff
@@ -292,6 +292,22 @@ def arithmetic(self):
                 o = LLLnode.from_list(['exp', left, right], typ=BaseType('num', new_unit), pos=getpos(self.expr))
             else:
                 raise TypeMismatchException('Only whole number exponents are supported', self.expr)
+        elif isinstance(self.expr.op, ast.FloorDiv):
+            if left.typ.positional or right.typ.positional:
+                raise TypeMismatchException("Cannot use divide and floor positional values!", self.expr)
+            new_unit = combine_units(left.typ.unit, right.typ.unit, div=True)
+            if ltyp == rtyp == 'num':
+                o = LLLnode.from_list(['sdiv', left, ['clamp_nonzero', right]], typ=BaseType('num', new_unit), pos=getpos(self.expr))
+            elif ltyp == rtyp == 'decimal':
+                o = LLLnode.from_list(['with', 'l', left, ['with', 'r', ['clamp_nonzero', right],
+                                            ['sdiv', 'l', 'r']]],
+                                            # ['mul', 'l', DECIMAL_DIVISOR]
```

Comment:
Why not just `['sdiv', left, ['clamp_nonzero', right]]` like when dividing `num` with `num`?

Reply:
@adhikasp Great catch, simplifying it.